emf/report_publisher/worker.py

```diff
@@ -0,0 +1,65 @@
+import uuid
+import config
+import logging
+from pathlib import Path
+from emf.common.integrations import rabbit, edx
+from emf.common.config_parser import parse_app_properties
+from emf.common.xslt_engine.saxonpy_api import validate_xml
+
+logger = logging.getLogger(__name__)
+parse_app_properties(globals(), config.paths.xslt_service.xslt)
+
+rabbit_service = rabbit.BlockingClient()
+edx_service = edx.EDX()
+
+schema_xml_path = Path(__file__).parent.joinpath('QAR_v2.6.1.xsd')
+
+
+def run_service(from_queue):
+
+    logger.info(f"Report publisher started")
+    rabbit_service.consume_start(queue=from_queue, callback=send_qar)
+
+
+def send_qar(channel, method, properties, body: str):
+
+    is_valid = properties.headers.get('is-valid')
+
+    if is_valid != 'True':
+        with open(schema_xml_path, 'rb') as file:
+            xsd_bytes = file.read()
```

Comment:
This validation part is reading XSD from filesystem every time, give the XSD to the function and load it once at startup